server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/webhook/WebhookPayload.java

```diff
@@ -0,0 +1,103 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.task.projectanalysis.webhook;
+
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.Date;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.sonar.api.ce.posttask.CeTask;
+import org.sonar.api.ce.posttask.PostProjectAnalysisTask;
+import org.sonar.api.ce.posttask.Project;
+import org.sonar.api.ce.posttask.QualityGate;
+import org.sonar.api.utils.text.JsonWriter;
+
+import static java.util.Objects.requireNonNull;
+
+public class WebhookPayload {
+
+  private final String projectKey;
+  private final String json;
+
+  public WebhookPayload(String projectKey, String json) {
+    this.projectKey = requireNonNull(projectKey);
+    this.json = requireNonNull(json);
+  }
+
+  public String getProjectKey() {
+    return projectKey;
+  }
+
+  public String toJson() {
+    return json;
+  }
+
+  public static WebhookPayload from(PostProjectAnalysisTask.ProjectAnalysis analysis) {
+    Writer string = new StringWriter();
+    JsonWriter writer = JsonWriter.of(string);
+    writer.beginObject();
+    writeTask(writer, analysis.getCeTask());
+    Optional<Date> analysisDate = analysis.getAnalysisDate();
+    if (analysisDate.isPresent()) {
+      writer.propDateTime("analysedAt", analysisDate.get());
+    }
+    writeProject(analysis, writer, analysis.getProject());
+    writeQualityGate(writer, analysis.getQualityGate());
+    writer.endObject().close();
+    return new WebhookPayload(analysis.getProject().getKey(), string.toString());
+  }
+
+  private static void writeTask(JsonWriter writer, CeTask ceTask) {
+    writer.prop("taskId", ceTask.getId());
+    writer.prop("status", ceTask.getStatus().toString());
+  }
+
+  private static void writeProject(PostProjectAnalysisTask.ProjectAnalysis analysis, JsonWriter writer, Project project) {
+    writer.name("project");
+    writer.beginObject();
+    writer.prop("key", project.getKey());
+    writer.prop("name", analysis.getProject().getName());
+    writer.endObject();
+  }
+
+  private static void writeQualityGate(JsonWriter writer, @Nullable QualityGate gate) {
+    if (gate != null) {
+      writer.name("qualityGate");
+      writer.beginObject();
+      writer.prop("name", gate.getName());
+      writer.prop("status", gate.getStatus().toString());
+      writer.name("conditions").beginArray();
+      for (QualityGate.Condition condition : gate.getConditions()) {
+        writer.beginObject();
+        writer.prop("metric", condition.getMetricKey());
+        writer.prop("operator", condition.getOperator().name());
+        writer.prop("value", condition.getValue());
+        writer.prop("status", condition.getStatus().name());
```

Comment:
I'm wrong. The stacktrace you provided does not relate to your explanation. 
Method `Condition#getValue()` has a not so intuitive behavior. Status NO_VALUE must be checked before.
Still I will ignore webkooks with missing name or URL.